### DISCUSSION TITLE
Help me to understand:- Lesson 6: Create the deposit collateral function

### DISCUSSION THREAD

**Discussion by 0xHuzaifa** [Cyfrin/foundry-full-course-cu#1372] [Q&A]
Hello,

I'm learning from Cyfrin updraft and I'm on **Advance Foundry Lesson 6: Create the deposit collateral function**. 

This part is confusing. If someone understands this part clearly please tell me.

**Thanks**

```
constructor(address[] memory tokenAddresses, address[] memory priceFeedAddresses, address dscAddress) {
        if (tokenAddresses.length != priceFeedAddresses.length) {
            revert DSCEngine__TokenAddressesAndPriceFeedAddressesAmountsDontMatch();
        }
        // These feeds will be the USD pairs
        // For example ETH / USD or MKR / USD
        for (uint256 i = 0; i < tokenAddresses.length; i++) {
            s_priceFeeds[tokenAddresses[i]] = priceFeedAddresses[i];
            s_collateralTokens.push(tokenAddresses[i]);
        }
        i_dsc = DecentralizedStableCoin(dscAddress);
    }
```

**Comment by note-xin** [comment]
dont know which part you need help with but here wht i understood

```solidity
constructor(address[] memory tokenAddresses, address[] memory priceFeedAddresses, address dscAddress) {
```
This constructor in the smart contract takes three parameters: an array of token addresses (`tokenAddresses`), an array of price feed addresses (`priceFeedAddresses`), and the address of a decentralized stable coin contract (`dscAddress`).

```solidity
if (tokenAddresses.length != priceFeedAddresses.length) {
    revert DSCEngine__TokenAddressesAndPriceFeedAddressesAmountsDontMatch();
}
```
This statement cheks if the lengths of the `tokenAddresses` array and the `priceFeedAddresses` array are unequal. It reverts the transaction with a custom error message using the `revert` keyword if they are unequal. The custom error message is generated by calling `DSCEngine__TokenAddressesAndPriceFeedAddressesAmountsDontMatch()` 

```solidity
for (uint256 i = 0; i < tokenAddresses.length; i++) {
    s_priceFeeds[tokenAddresses[i]] = priceFeedAddresses[i];
    s_collateralTokens.push(tokenAddresses[i]);
}
```
for the loop it goes through over each element in the `tokenAddresses` array. For each token address, it assigns the corresponding price feed address from the `priceFeedAddresses` array to the `s_priceFeeds` mapping, where the token address is the key and the price feed address is the value.
Additionally, it adds the token address to the `s_collateralTokens` array.

```solidity
i_dsc = DecentralizedStableCoin(dscAddress);
```
This line assigns the `dscAddress` to the `i_dsc` variable, casting it to the `DecentralizedStableCoin` contract type. 

summary
this constructor  initializes a mapping of token addresses to price feed addresses
checks that the lengths of two arrays are equal and assigns the address of a decentralized stable coin contract to a variable.

don't trust wht i said completely as im new to solidity as well. have fun coding
peace and blessing to you friend
